### PR TITLE
feat: display a warning when there is no latest tag for image to pull

### LINKED ISSUE #8927: pulling/searching images without latest tag
### Is your enhancement related to a problem? Please describe

now, it's possible to get typeahead completion on images and on the tags

### Describe the solution you'd like

I would like to be notified if the image has not a latest tag (it does not exists a lot but it's still exist so I should get a warning if image is recognized/found and we see the tag does not exist so I should pick up a tag)

### Describe alternatives you've considered

_No response_

### Additional context

_No response_

### PR DESCRIPTION
**Pull request by feloy**
Signed-off-by: Philippe Martin <phmartin@redhat.com>

### What does this PR do?

Display a warning when no latest tag for image to pull.

### Screenshot / video of UI

https://github.com/user-attachments/assets/01abd98b-f992-4d03-b1d2-9630ab210b6f

### What issues does this PR fix or reference?

Fixes #8927 

### How to test this PR?

Try to pull an image with or without "latest" tag (for example quay.io/phmartin/* images mostly don't have it)

- [x] Tests are covering the bug fix or the new feature

### CODE REVIEW

File: packages/renderer/src/lib/image/PullImage.svelte [hunk 1]
```diff
@@ -204,6 +205,29 @@ async function searchImages(value: string): Promise<string[]> {
...
+  }
+  try {
+    let image = imageToPull;
+    if (image.startsWith(DOCKER_PREFIX + '/')) {
+      image = image.slice(DOCKER_PREFIX.length + 1);
+    }
+    const tags = await window.listImageTagsInRegistry({ image });
+    const latestFound = tags.includes('latest');
+    if (!latestFound) {
+      latestTagMessage = '"latest" tag not found. You can search a tag by typing ":"';
```
Comment (benoitf) [inline packages/renderer/src/lib/image/PullImage.svelte]: question: wondering about replacing typing word by appending or adding

**Review by benoitf — APPROVED** [review]
image

works as expected 👍

File: packages/renderer/src/lib/image/PullImage.svelte [hunk 2]
```diff
@@ -204,6 +205,29 @@ async function searchImages(value: string): Promise<string[]> {
...
+let latestTagMessage: string | undefined = undefined;
+async function searchLatestTag(): Promise<void> {
+  if (imageNameIsInvalid || !imageToPull || imageToPull.includes(':')) {
+    latestTagMessage = undefined;
+    return;
+  }
+  try {
+    let image = imageToPull;
+    if (image.startsWith(DOCKER_PREFIX + '/')) {
+      image = image.slice(DOCKER_PREFIX.length + 1);
```
Comment (benoitf) [inline packages/renderer/src/lib/image/PullImage.svelte]: question: should it be another constant ?

so no need to add a new / each time or to add +1 as well

File: packages/renderer/src/lib/image/PullImage.svelte (hunk 1 above)
Comment (feloy) [reply packages/renderer/src/lib/image/PullImage.svelte]: Appending seems more accurate to me. I'll change

File: packages/renderer/src/lib/image/PullImage.svelte (hunk 2 above)
Comment (feloy) [reply packages/renderer/src/lib/image/PullImage.svelte]: I'm not sure to find a constant name more explicit than these formulas

Comment (benoitf) [reply packages/renderer/src/lib/image/PullImage.svelte]: top of the file near const DOCKER_PREFIX = 'docker.io';
```ts
const DOCKER_PREFIX = 'docker.io';
const DOCKER_PREFIX_WITH_SLASH = DOCKER_PREFIX + '/'
```

and then here

```ts
if (image.startsWith(DOCKER_PREFIX_WITH_SLASH)) {
  image = image.slice(DOCKER_PREFIX_WITH_SLASH.length);
}
```

**Review by dgolovin — APPROVED** [review]
Resolved conflicts and checked locally.
LGTM.

**Pull request merged**